add viscosity profile that is more consistent with original Steinberg…

jdannberg opened the pull request:
…er and Calderwood formulation

This is based on a conversation with Bernhard Steinberger, who sent me this file, noting that it is more consistent with the profiles computed in his paper. 

I have also updated the cookbook to show the difference between the profiles. 

### For all pull requests:

* [x] I have followed the [instructions for indenting my code](../blob/main/CONTRIBUTING.md#making-aspect-better).

### For new features/models or changes of existing features:

* [x] I have tested my new feature locally to ensure it is correct.
* [x] I have [created a testcase](https://aspect-documentation.readthedocs.io/en/latest/user/extending/testing/writing-tests.html) for the new feature/benchmark in the [tests/](../blob/main/tests/) directory.
* [x] I have added a changelog entry in the [doc/modules/changes](../blob/main/doc/modules/changes) directory that will inform other users of my change.

gassmoeller commented on cookbooks/steinberger/doc/steinberger.md:
```diff
@@ -247,6 +247,37 @@ Viscosity profile based on the original {cite:t}`steinberger:calderwood:2006` fo
+optimize them based on observational data. During this optimization, the different
+layers that make up the profiles (usually lithosphere, upper mantle, transition
+zone, and lower mantle) are shifted left or right relative to each other.
+The figures that show these optizimed profiles in the paper represent them as
```
```suggestion
The figures that show these optimized profiles in the paper represent them as
```

gassmoeller commented on cookbooks/steinberger/doc/steinberger.md:
```diff
@@ -247,6 +247,37 @@ Viscosity profile based on the original {cite:t}`steinberger:calderwood:2006` fo
+layers that make up the profiles (usually lithosphere, upper mantle, transition
+zone, and lower mantle) are shifted left or right relative to each other.
+The figures that show these optizimed profiles in the paper represent them as
+piece-wise constant within 22 layers, and the profiles shown in Figures
```
```suggestion
piece-wise constant values within 22 layers, and the profiles shown in Figures
```

gassmoeller commented on doc/modules/changes/20230823_dannberg:
```diff
@@ -0,0 +1,5 @@
+New: There is now a new viscosity profile data file that
+is more consistent with how the profile is computed in
+the original Steinberger and Calderwood paper.
```
```suggestion
the original Steinberger and Calderwood (2006) paper.
```

gassmoeller reviewed (commented):
Thank you for fixing this! The PR looks good, I just marked some small possible improvements. As far as I understand we prefer the new profile as a more faithful representation of the original paper. Should we make this more clear in the cookbook description so that users will preferentially use the new profile? Maybe make the new profile the default in the material model? (this would change user models, so maybe not?) Not sure what the best path is, let me know what you think.

jdannberg commented:
I addressed your comments. 
For the default value, I checked this when making the PR, and the current default is `set Radial viscosity file name = radial-visc.txt` so it's still the profile with the boundary layers at the top and bottom. That's why I thought we probably do not want to change it. 

But I added a sentence that we recommend the new profile.
